test/integration/scheduler/framework_test.go

```diff
@@ -1885,10 +1884,14 @@ func TestPreemptWithPermitPlugin(t *testing.T) {
 	// }
 
 	if err := wait.Poll(200*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
-		_, err := getPod(testCtx.ClientSet, waitingPod.Name, waitingPod.Namespace)
-		return apierrors.IsNotFound(err), nil
+		w := false
+		permitPlugin.fh.IterateOverWaitingPods(func(wp framework.WaitingPod) { w = true })
+		if _, err := getPod(testCtx.ClientSet, waitingPod.Name, waitingPod.Namespace); err != nil {
+			return false, nil
+		}
```

Comment:
I meant to move this to L1896 as we don't need to run the API check inside this loop - only check it once.

Reply:
done